### ISSUE TITLE
咨询数据集

### ISSUE THREAD

**Issue by wang-shuibin** [aikunyi/FourierGNN#6]
你好，能提供几个数据集嘛（.CSV格式的）

**Comment by a1032077316** [comment]
同求

**Comment by xiatian815** [comment]
[ECG_data.csv](https://github.com/aikunyi/FourierGNN/files/13737648/ECG_data.csv)
[metr.csv](https://github.com/aikunyi/FourierGNN/files/13737649/metr.csv)
这些数据我跑出来的效果很差，不知道是不是因为显卡不行的原因

**Comment by xuexue49** [comment]
ecg的数据集效果是不错的

> [ECG_data.csv](https://github.com/aikunyi/FourierGNN/files/13737648/ECG_data.csv) [metr.csv](https://github.com/aikunyi/FourierGNN/files/13737649/metr.csv) 这些数据我跑出来的效果很差，不知道是不是因为显卡不行的原因

**Comment by xiatian815** [comment]
![QQ图片20231222144221](https://github.com/aikunyi/FourierGNN/assets/57095039/906625f1-e34b-4ce4-8416-bce145837ace)
感觉不太理想

**Comment by wang-shuibin** [comment]
我也跑出来效果不理想

**Comment by aikunyi** [comment]
Thank you for your interest in our work. Maybe you should execute the test function, not just train

**Comment by wang-shuibin** [comment]
okok

**Comment by xiatian815** [comment]
> 感谢您对我们工作的兴趣。也许你应该执行测试函数
哪个py文件是进行测试的呢

**Comment by lurenyi233** [comment]
同想请教一下，我在ECG数据集上test的效果比较差，这可能是什么原因呢？
![11](https://github.com/aikunyi/FourierGNN/assets/33956086/6f91b780-f47a-4ac0-96b3-3cc4d20c1828)

下面是我用的参数，十分感谢！！
parser = argparse.ArgumentParser(description='fourier graph network for multivariate time series forecasting')
parser.add_argument('--data', type=str, default='ECG', help='data set')
parser.add_argument('--feature_size', type=int, default='140', help='feature size')
parser.add_argument('--seq_length', type=int, default=12, help='inout length')
parser.add_argument('--pre_length', type=int, default=12, help='predict length')
parser.add_argument('--embed_size', type=int, default=128, help='hidden dimensions')
parser.add_argument('--hidden_size', type=int, default=256, help='hidden dimensions')
parser.add_argument('--train_epochs', type=int, default=100, help='train epochs')
parser.add_argument('--batch_size', type=int, default=4, help='input data batch size')
parser.add_argument('--learning_rate', type=float, default=0.00001, help='optimizer learning rate')
parser.add_argument('--exponential_decay_step', type=int, default=5)
parser.add_argument('--validate_freq', type=int, default=1)
parser.add_argument('--early_stop', type=bool, default=False)
parser.add_argument('--decay_rate', type=float, default=0.5)
parser.add_argument('--train_ratio', type=float, default=0.7)
parser.add_argument('--val_ratio', type=float, default=0.2)
parser.add_argument('--device', type=str, default='cuda:0', help='device')

**Comment by timc12** [comment]
data leakage很大程度影响了ECG数据集结果，没有https://github.com/aikunyi/FourierGNN/pull/11 前MAPE大概20%多，没了data leakage就大概50%多

**Comment by aikunyi** [comment]
已经修改了data_loader.py文件，建议重跑下，我们没有观测到您说的这种情况

**Comment by day338** [comment]
> > 感谢您对我们工作的兴趣。也许你应该执行测试函数
> > 哪个py文件是进行测试的呢

同问，哪个文件是进行测试的，楼主知道了吗

**Comment by zhangweish** [comment]
请问数据集在哪里下载呢？

**Comment by Zrmmm** [comment]
> [ECG_data.csv ](https://github.com/aikunyi/FourierGNN/files/13737648/ECG_data.csv) [metr.csv](https://github.com/aikunyi/FourierGNN/files/13737649/metr.csv) 这些数据我跑出来的效果很差，不知道是不是因为显卡不行的原因
请问这个METR.CSV文件是论文中那个METR-LA数据集吗,为什么我用这个文件跑的效果比论文中好?

**Comment by Solaaaaa** [comment]
> > 感谢您对我们工作的兴趣。也许你应该执行测试函数
> > 哪个py文件是进行测试的呢

你问的可能是测试函数，在源文件 main.py 中没有执行test（）函数，你可以在最后加上一句test（）来执行test函数。
![屏幕截图 2024-07-19 174216](https://github.com/user-attachments/assets/11c84e7b-8014-4229-84df-eb1e8bdf9d6f)
另外，main.py 中的test（）函数中的这条语句：
model = load_model（result_test_file， 48）
可能要修改为：
model = load_model（result_test_file， epoch）
希望能够帮到你！

**Comment by Solaaaaa** [comment]
> 请问数据集在哪里下载呢？
README中的论文里有，以下是论文里的几个数据集的下载链接：
https://www.nrel.gov/grid/solar-power-data.html 
http://www.timeseriesclassification.com/description.php?Dataset=ECG5000 https://archive.ics.uci.edu/ml/datasets/ElectricityLoadDiagrams20112014 
https://github.com/CSSEGISandData/COVID-19 
https://github.com/liyaguang/DCRNN

**Comment by fftfcd** [comment]
![微信图片_20240801141240](https://github.com/user-attachments/assets/3ae72cbf-3815-4ca2-8a46-f8dee40f5cf4)
不大理想

**Comment by Solaaaaa** [comment]
> ![微信图片_20240801141240](https://private-user-images.githubusercontent.com/137707039/354102451-3ae72cbf-3815-4ca2-8a46-f8dee40f5cf4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2ODQzMzYsIm5iZiI6MTcyMjY4NDAzNiwicGF0aCI6Ii8xMzc3MDcwMzkvMzU0MTAyNDUxLTNhZTcyY2JmLTM4MTUtNGNhMi04YTQ2LWY4ZGVlNDBmNWNmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwM1QxMTIwMzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ODRiODBlZDI0Nzg3MmExNmZhZjRjZWFkZGFkOTU2NDkxNDA0YzM0NmRhMWNlNzQ2MGIyNjMzOWYwYmRjZTNiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._piLOhqx4enUG94Yq9xZYVgwFuEKJDYKfdPy1bm5Sc8)不大理想

我猜你跑的应该是ECG数据集，以下是我跑ECG的结果：
![image](https://github.com/user-attachments/assets/1f74e97b-9f4d-4229-9c93-785afa0b6817)
我怀疑我们差这么多的原因可能是你使用了https://github.com/aikunyi/FourierGNN/files/13737648/ECG_data.csv    这个ECG数据集，请注意这个数据集可能是错的，论文中描述的ECG数据集的链接应该为  http://www.timeseriesclassification.com/description.php?Dataset=ECG5000
但是我跑出来的结果还是和论文结果有不小的差距。论文中MAPE为10.97%，MAE为0.052，RMSE为0.078，以下是我的参数，我想请教一下作者原因，十分感谢！！
parser = argparse.ArgumentParser(description='fourier graph network for multivariate time series forecasting')
parser.add_argument('--data', type=str, default='ECG', help='data set')  #ECG
parser.add_argument('--feature_size', type=int, default='140', help='feature size') #140
parser.add_argument('--seq_length', type=int, default=12, help='inout length')   #每个窗口的T是固定的
parser.add_argument('--pre_length', type=int, default=12, help='predict length')
parser.add_argument('--embed_size', type=int, default=128, help='hidden dimensions')
parser.add_argument('--hidden_size', type=int, default=256, help='hidden dimensions')
parser.add_argument('--train_epochs', type=int, default=100, help='train epochs')  #100
parser.add_argument('--batch_size', type=int, default=32, help='input data batch size')
parser.add_argument('--learning_rate', type=float, default=0.00001, help='optimizer learning rate')
parser.add_argument('--exponential_decay_step', type=int, default=5)
parser.add_argument('--validate_freq', type=int, default=1)
parser.add_argument('--early_stop', type=bool, default=False)
parser.add_argument('--decay_rate', type=float, default=0.5)
parser.add_argument('--train_ratio', type=float, default=0.7)
parser.add_argument('--val_ratio', type=float, default=0.2)
parser.add_argument('--device', type=str, default='cuda:0', help='device')
